python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py

```diff
@@ -250,10 +271,19 @@ def __init__(
         self._tensor_attr_cls = CuGraphTensorAttr
         self._tensor_attr_dict = defaultdict(list)
 
-        # Infer number of edges from the edge index dict
-        num_edges_dict = {
-            pyg_can_edge_type: len(ei[0]) for pyg_can_edge_type, ei in G.items()
-        }
+        construct_graph = True
+        if isinstance(next(iter(G.values())), int):
+            # User has passed in the number of edges 
+            # (not the actual edge index), so the number of edges
+            # does not need to be counted.
+            num_edges_dict = dict(G) # make sure the cugraph store owns this dict
```

Comment:
What does `make sure the cugraph store owns this dict` ensure.

Reply:
It's copying the dictionary so that if it is changed later, the version saved in the `CuGraphStore` object is not affected.